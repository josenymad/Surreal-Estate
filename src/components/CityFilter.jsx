import React from "react";
import { Link } from "react-router-dom";
import "../styles/city-filter.css";

const CityFilter = ({ setCity }) => {
  const cities = ["Manchester", "Leeds", "Sheffield", "Liverpool"];

  return (
    <div className="CityFilter">
      {cities.map((city) => {
        return (
          <Link
            key={city}
            className="city-filter__link"
            to={`/?query={"city": "${city}" }`}
            onClick={setCity(city)}
          >
            {city}
          </Link>
        );
      })}
    </div>
  );
};

export default CityFilter;
