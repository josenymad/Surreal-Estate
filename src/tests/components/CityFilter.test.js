import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CityFilter from "../../components/CityFilter";

describe("City Filter", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CityFilter />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders 4 city links", () => {
    render(
      <MemoryRouter>
        <CityFilter />
      </MemoryRouter>
    );

    const links = screen.getAllByRole("link");

    expect(links).toHaveLength(4);
  });
});
