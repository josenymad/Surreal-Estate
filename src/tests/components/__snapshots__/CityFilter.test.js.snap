// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City Filter renders correctly 1`] = `
<DocumentFragment>
  <div
    class="CityFilter"
  >
    <a
      class="city-filter__link"
      href="/?query={\\"city\\": \\"Manchester\\" }"
    >
      Manchester
    </a>
    <a
      class="city-filter__link"
      href="/?query={\\"city\\": \\"Leeds\\" }"
    >
      Leeds
    </a>
    <a
      class="city-filter__link"
      href="/?query={\\"city\\": \\"Sheffield\\" }"
    >
      Sheffield
    </a>
    <a
      class="city-filter__link"
      href="/?query={\\"city\\": \\"Liverpool\\" }"
    >
      Liverpool
    </a>
  </div>
</DocumentFragment>
`;
